tests/testImageDecorrelation.py

```diff
@@ -190,65 +182,124 @@ def setUp(self):
-        sig_match = np.sqrt((psf2_sig**2. - psf1_sig**2.))
-        self.assertClose(sig_match, np.sqrt((self.psf2_sigma**2. - self.psf1_sigma**2.)), rtol=1e-5)
-        #matchingKernel = measAlg.SingleGaussianPsf(31, 31, sig_match)
+        sig_match = np.sqrt((psf1_sig**2. - psf2_sig**2.))
+        self.assertClose(sig_match, np.sqrt((self.psf1_sigma**2. - self.psf2_sigma**2.)), rtol=1e-5)
+        # mKernel = measAlg.SingleGaussianPsf(31, 31, sig_match)
         x0 = np.arange(-16, 16, 1)
         y0 = x0.copy()
         x0im, y0im = np.meshgrid(x0, y0)
-        matchingKernel = singleGaussian2d(x0im, y0im, 0., 0., sigma_x=sig_match, sigma_y=sig_match)
+        matchingKernel = singleGaussian2d(x0im, y0im, -1., -1., sigma_x=sig_match, sigma_y=sig_match)
         kernelImg = afwImage.ImageD(matchingKernel.shape[0], matchingKernel.shape[1])
         kernelImg.getArray()[:, :] = matchingKernel
         mKernel = afwMath.FixedKernel(kernelImg)
-        mKernel = measAlg.KernelPsf(mKernel)
 
         # Create the matched template by convolving the template with the matchingKernel
-        from scipy.ndimage.filters import convolve
-        im1 = self.im1ex.getMaskedImage().getImage().getArray()
-        matched_im1 = convolve(im1, matchingKernel, mode='constant')
-        matched_im1ex = self.im1ex.clone()
-        matched_im1ex.getMaskedImage().getImage().getArray()[:, :] = matched_im1
+        matched_im2ex = self.im2ex.clone()
+        convCntrl = afwMath.ConvolutionControl(False, True, 0)
+        afwMath.convolve(matched_im2ex.getMaskedImage(), self.im2ex.getMaskedImage(), mKernel, convCntrl)
 
         # Expected (ideal) variance of difference image
-        expected_var = self.sig1**2 + self.sig2**2
+        expected_var = self.var1 + self.var2
         print('Expected variance:', expected_var)
 
-        im2 = self.im2ex.getMaskedImage().getImage().getArray()
-        print(np.nan_to_num(matched_im1 - im2).var())
+        statObj = afwMath.makeStatistics(self.im1ex.getMaskedImage(), afwMath.VARIANCECLIP,
+                                         self.statsControl)
+        var = statObj.getValue(afwMath.VARIANCECLIP)
+        statObj = afwMath.makeStatistics(self.im2ex.getMaskedImage(), afwMath.VARIANCECLIP,
+                                         self.statsControl)
+        var = statObj.getValue(afwMath.VARIANCECLIP)
+
+        im1 = self.im1ex.getMaskedImage().getImage().getArray()
+        matched_im2 = matched_im2ex.getMaskedImage().getImage().getArray()
         # Uncorrected diffim - variance is wrong (too low)
-        self.assertNotClose(np.nan_to_num(matched_im1 - im2).var(), expected_var, rtol=0.1)
-        # In fact, it should have an (incorrect) variance that is close to the variance of the science image.
-        self.assertClose((matched_im1 - im2)[~np.isnan(matched_im1 - im2)].var(), self.sig2**2., rtol=0.1)
+        self.assertNotClose(np.nan_to_num(im1 - matched_im2).var(), expected_var, rtol=0.1)
```

Comment:
Can you quantify this test better? Instead of being "not close" is it that the first value is much less than the second, or something? That would make the test more robust.